### PR TITLE
Update operator_kubecluster.rst

### PR DESCRIPTION
**Pull request by jadeidev**
update docs for starting KubeCluster from pod in the cluster

### CODE REVIEW

**Comment by jadeidev** [comment]
Closes #582

File: doc/source/operator_kubecluster.rst [hunk 1]
```diff
@@ -136,6 +136,62 @@ The ``cluster.add_worker_group()`` method also supports passing a ``custom_spec`
...
+       verbs: ["*"]
+
+     - apiGroups: 
+       - ""  # indicates the core API group
+       resources: [services]
+       verbs: ["*"]
+
+     - apiGroups: ["", events.k8s.io]
+       resources: [events]
+       verbs: ["*"]
```
Comment (jacobtomlinson) [inline doc/source/operator_kubecluster.rst:172]: I don't think we should need any of these as the controller handles creating all of the objects.

**Review by jacobtomlinson — COMMENTED** [review]
Many thanks for this, totally agree this is a good thing to document.

I don't think we need all of the permissions you have listed though. AFAIK we need the full `kubernetes.dask.org` resources and verbs and then a few get/list operations to get things like logs and statuses.

Could you double-check the permissions you have included here and remove any that are unnecessary?

File: doc/source/operator_kubecluster.rst (hunk 1 above)
Comment (jadeidev) [reply doc/source/operator_kubecluster.rst]: @jacobtomlinson I have updated the verbs and resources needed. I believe these are the minimum requirements to get things going.

File: doc/source/operator_kubecluster.rst [hunk 2]
```diff
@@ -136,6 +136,55 @@ The ``cluster.add_worker_group()`` method also supports passing a ``custom_spec`
...
+   metadata:
+     name: dask-cluster-role
+   rules:
+     # Application: watching & handling for the custom resource we declare.
+     - apiGroups: [kubernetes.dask.org]
+       resources: [daskclusters, daskworkergroups, daskworkergroups/scale, daskjobs, daskautoscalers]
+       verbs: [get, list, watch, patch, create, delete]
+
+     # Application: other resources it produces and manipulates.
+     # Here, we create/delete Pods.
```
Comment (jacobtomlinson) [inline doc/source/operator_kubecluster.rst:159]:
```suggestion
     # Application: other resources it needs to watch and get information from.
```

**Comment by jacobtomlinson** [comment]
Many thanks!

**Pull request merged**